internal/oasutil/servers.go

```diff
@@ -156,8 +157,10 @@ func (p *serverUrlParser) extractValueBetweenBraces() (serverVariable, error) {
 				return serverVariable{}, ErrInvalidVariableName
 			}
 
-			if _, err := regexp.Compile(pattern); err != nil {
+			if re, err := regexp.Compile(pattern); err != nil {
 				return serverVariable{}, errpack.Wrap(ErrInvalidPattern, "failed to compile pattern")
+			} else if hasCaptureGroups(re) {
```

Comment:
curious, why the extra one line function (Doesn't need to be changed, just wondered if there is a future reason)